### PR TITLE
Tests on separate db

### PR DESCRIPTION
**Pull request by pstaabp**
This switches the unit tests for both the database and mojolicious to a database separate than the one that will be run in regular circumstances.  This helps with the separation of the testing code as well addressing issue #41. A few things to note:
1. There is now a `conf/ww3-dev.dist.yml` file for development variables including the testing database information.
2. Copy `conf/ww3-dev.dist.yml `to `conf/ww3-dev.yml`
3. Run `perl t/db/sample_data/build_db.pl`
4. Run all tests with `prove -r t` from the main webwork3 directory. 

Test with both a sqlite db and some sql database (mysql or mariadb) and on the latter, you'll need to create that database.

### CODE REVIEW

**Review by drgrice1 — CHANGES_REQUESTED** [review]
There are some things that could be done better here.  See my comments.

I didn't comment on the addition of the super-linter.log file as I already deleted that in the dependent pull request.  You can leave the Dumpers also, as @drdrew42 removed those in the dependent pull request.  There may be some other unnecessary `use` statements, but we will deal with those later.

Make the changes that I commented on that don't say to leave for later, and then we can merge this and move on.

File: conf/webwork3.dist.yml [hunk 1]
```diff
@@ -3,10 +3,6 @@ secrets:
   - 3cdf63327fcf77deaed1d200df4b9fee66af2326
 webwork3_home: /opt/webwork/webwork3
 
-# If ignore_permissions is set to true, all routes can be executed.
-# This should only be used in development.
-ignore_permissions: true
```
[the remark this answers is not in the document]
Comment (pstaabp) [reply conf/webwork3.dist.yml:8]: There are some routes with permissions on the Mojolicious side of the test and the idea is to also run those (but they aren't set up for automatic testing).  I don't quite remember how I intended to use this to run the tests, but will delete these lines and rethink how to handle permissions testing.

**Comment by pstaabp** [comment]
Merging main back in broke some things.  Not sure I have time to fix before our meeting today. No need to look yet.

**Comment by drdrew42** [comment]
@pstaabp @drgrice1 this is now passing, have we covered all the feedback?

**Comment by drgrice1** [comment]
I think that you misinterpreted what I meant with the `test_` prefix.  What I meant was to remove that prefix, not to remove the username and password settings from the `ww3-dev.dist.yml` file.  Those will need to go back in.  Without them tests will fail if you are trying to run tests with a database (mysql, mariadb, postgres, etc) that needs a username and password.  The tests now only work for the sqlite database that doesn't need the username and password.  The prefix should also be removed from the `database_dsn`. 

The point is really that the ww3-dev.yml file should really be much the same as the main yml config file, but with the settings needed for tests and perhaps stripped down (remove things like the hypnotoad settings that tests don't need).

Other than that it looks like you addressed the main concerns.

**Comment by drgrice1** [comment]
Also, the bit about cleaning up the config file loading needs to be done in the mojolicious test files.  That should be identical to the other test files with the exception of adding clone to the LoadFile part.

**Comment by drgrice1** [comment]
Oh wait. You only did that clean up in the one place that I put the comment.  I said to do it in `all` of the test files.

**Comment by drdrew42** [comment]
I've gone ahead and copy-pasted the updated LoadFiles snippet across the test files.

**Comment by drgrice1** [comment]
I added a pull request to this one that implements the removal of the `_test` prefix.  I also made the paths relative in the webwork3.dist.yml file as you did in the ww3-dev.dist.yml file.  So now a simple `cp` can be used in the unit test workflow file instead of an `sed` statement.

Merge that, and then this pull request can be merged.

**Comment by pstaabp** [comment]
Merged that i.  I was sloppy with the previous set of changes.  A case of taking enough time and care.  Thanks for the fixes.

**Comment by drgrice1** [comment]
Okay.  We can move on to the next pull request.

**Pull request merged**